src/ExternalEvents.php

```diff
@@ -26,45 +26,54 @@ public static function publish(Message $class): void
                 )
             )
         );
-        $message    = [
+
+        $message = [
             'data' => $class->serializeToJsonString(),
         ];
 
-        publish($routingKey, $message);
-    }
-
-    /**
-     * @throws InvalidMessageException
-     */
-    public static function decode(string $expectedEvent, string $message): mixed
-    {
-        try {
-            $event = new $expectedEvent();
-            $event->mergeFromJsonString($message);
-
-            return $event;
-        } catch (Exception) {
-            throw new InvalidMessageException(
-                "The message is not a valid {$expectedEvent} message"
-            );
+        if ($headers) {
+            $message['headers'] = $headers;
```

Comment:
I think we could always send the headers even if they are empty. With that we will  get an uniform message with the same structure.